pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java

```diff
@@ -251,8 +251,10 @@ public static ColumnMetadataImpl fromPropertiesConfiguration(String column, Prop
     // NOTE: Use getProperty() instead of getString() to avoid variable substitution ('${anotherKey}'), which can cause
     //       problem for special values such as '$${' where the first '$' is identified as escape character.
     // TODO: Use getProperty() for other properties as well to avoid the overhead of variable substitution
-    String minString = (String) config.getProperty(Column.getKeyFor(column, Column.MIN_VALUE));
-    String maxString = (String) config.getProperty(Column.getKeyFor(column, Column.MAX_VALUE));
+    String minString = StringEscapeUtils.unescapeJava((String)
```

Comment:
This can cause problem when the original value has `\` because we didn't call `escapeJava()` when writing the value. Find this issue in this test [run](https://github.com/apache/pinot/actions/runs/5691584661/job/15427035722?pr=11206). I think we should just use `escapeJava()` to solve the whitespace and unicode issue